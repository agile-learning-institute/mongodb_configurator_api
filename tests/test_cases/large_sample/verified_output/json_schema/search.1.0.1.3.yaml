description: A search index that is used to support a elastic search polymorphic query
  service
type: object
properties:
  collection_name:
    description: The name of the collection
    type: string
    pattern: ^\S{1,40}$
  collection_id:
    description: The unique identifier for this source document
    type: string
    pattern: ^[0-9a-fA-F]{24}$
required:
- collection_name
- collection_id
additionalProperties: false
