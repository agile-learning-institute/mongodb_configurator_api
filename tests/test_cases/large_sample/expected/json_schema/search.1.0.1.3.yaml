title: search
description: A search index that is used to support a elastic search polymorphic query service
type: object
properties:
  collection_name:
    description: The name of the collection
    type: string
    pattern: "^[a-zA-Z0-9_-]{4,40}$"
  collection_id:
    description: The unique identifier for this source document
    type: string
    pattern: "^[0-9a-fA-F]{24}$"
  media:
    title: Media
    description: A media item in the system
    type: object
    properties:
      _id:
        description: The unique identifier for the media
        type: string
        pattern: "^[0-9a-fA-F]{24}$"
      title:
        description: The title of the media
        type: string
        pattern: "^[a-zA-Z0-9\\s.,!?-]{4,255}$"
      type:
        description: The type of media
        type: string
        enum:
          - movie
          - tv_show
          - documentary
          - short
      status:
        description: The current status of the media
        type: string
        enum:
          - draft
          - published
          - archived
      last_saved:
        title: Breadcrumb
        description: When the media was last updated
        type: object
        properties:
          from_ip:
            description: Http Request remote IP address
            type: string
            pattern: "^[a-zA-Z0-9_-]{4,40}$"
          by_user:
            description: ID Of User
            type: string
            pattern: "^[a-zA-Z0-9_-]{4,40}$"
          at_time:
            description: The date-time when last updated
            type: string
            format: date-time
          correlation_id:
            description: The logging correlation ID of the update transaction
            type: string
            pattern: "^[a-zA-Z0-9_-]{4,40}$"
        required:
          - from_ip
          - by_user
          - at_time
          - correlation_id
        additionalProperties: false
      tags:
        description: Tags associated with the media
        type: array
        items:
          type: string
          enum:
            - action
            - comedy
            - drama
            - sci_fi
            - documentary
      metadata:
        description: Additional metadata about the media
        type: object
        properties:
          duration: 
            description: Duration in minutes
            type: number
            minimum: 1
            multipleOf: 1
          format:
            description: Media format
            type: string
            enum:
              - dvd
              - bluray
              - digital
              - streaming
          quality:
            description: Quality rating
            type: string
            enum:
              - sd
              - hd
              - uhd
        additionalProperties: false
    required:
      - _id
      - title
      - status
      - last_saved
    additionalProperties: false
  organization:
    title: Organization
    description: An organization in the system
    type: object
    properties:
      _id:
        type: string
        description: The unique identifier for the organization
        pattern: "^[0-9a-fA-F]{24}$"
      name:
        description: The organization name
        type: string
        pattern: "^[a-zA-Z0-9\\s.,!?-]{4,255}$"
      status:
        description: The current status of the organization
        type: string
        enum:
          - draft
          - active
          - archived
      last_saved:
        title: Breadcrumb
        description: When the organization document was last updated
        type: object
        properties:
          from_ip:
            description: Http Request remote IP address
            type: string
            pattern: "^[a-zA-Z0-9_-]{4,40}$"
          by_user:
            description: ID Of User
            type: string
            pattern: "^[a-zA-Z0-9_-]{4,40}$"
          at_time:
            description: The date-time when last updated
            type: string
            format: date-time
          correlation_id:
            description: The logging correlation ID of the update transaction
            type: string
            pattern: "^[a-zA-Z0-9_-]{4,40}$"
        required:
          - from_ip
          - by_user
          - at_time
          - correlation_id
        additionalProperties: false
      users:
        description: Users associated with this organization
        type: array
        items:
          description: A user identifier
          type: string
          pattern: "^[0-9a-fA-F]{24}$"
      website:
        description: Organization website
        type: string
        pattern: "^https?://[\\w\\d\\-]+(\\.[\\w\\d\\-]+)+([\\w\\d\\-._~:/?#\\[\\]@!$&'()*+,;=]*)?$"
      home_address:
        title: Street Address
        description: Organization home address
        type: object
        properties:
          street:
            description: Street address
            type: string
            pattern: "^[a-zA-Z0-9\\s.,!?-]{4,255}$"
          city:
            description: City
            type: string
            pattern: "^[a-zA-Z0-9_-]{4,40}$"
          state:
            description: State or province
            type: string
            pattern: "^[a-zA-Z0-9_-]{4,40}$"
          postal_code:
            description: Postal code
            type: string
            pattern: "^[a-zA-Z0-9_-]{4,40}$"
        required:
          - postal_code
        additionalProperties: false
    required:
      - _id
      - name
      - status
      - last_saved
    additionalProperties: false
  user:
    title: User Collection
    description: A user collection for testing the schema system
    type: object
    properties:
      _id:
        type: string
        description: The unique identifier for a user
        pattern: "^[0-9a-fA-F]{24}$"
      user_name:
        type: string
        description: Username
        pattern: "^[a-zA-Z0-9_-]{4,40}$"
      full_name:
        type: string
        description: Users Full Name
        pattern: "^[a-zA-Z0-9\\s.,!?-]{4,255}$"
      status:
        type: string
        description: The status
        enum:
          - draft
          - active
          - archived
      categories:
        type: array
        description: A users list of categorized tags
        items:
          type: object
          description: A user category
          properties:
            name:
              type: string
              description: Category Name assigned by the user
              pattern: "^[a-zA-Z0-9_-]{4,40}$"
            category:
              type: string
              description: The category type assigned by the user
              enum:
                - work
                - personal
                - project
                - reference
            tags:
              type: array
              description: A list of enumerated values assigned by the user
              items:
                type: string
                enum:
                  - urgent
                  - important
                  - normal
                  - low
                  - completed
                  - in_progress
                  - blocked
                  - review
          required:
            - name
            - category
            - tags
          additionalProperties: false
      email:
        type: string
        description: The person's email address
        pattern: "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$"
      phone:
        type: string
        description: The person's phone number
        pattern: "^\\+1[2-9][0-9]{9}$"
      last_saved:
        title: Breadcrumb
        description: The last time this document was saved
        type: object
        properties:
          from_ip:
            description: Http Request remote IP address
            type: string
            pattern: "^[a-zA-Z0-9_-]{4,40}$"
          by_user:
            description: ID Of User
            type: string
            pattern: "^[a-zA-Z0-9_-]{4,40}$"
          at_time:
            description: The date-time when last updated
            type: string
            format: date-time
          correlation_id:
            description: The logging correlation ID of the update transaction
            type: string
            pattern: "^[a-zA-Z0-9_-]{4,40}$" 
        required:
          - from_ip
          - by_user
          - at_time
          - correlation_id
        additionalProperties: false
    required:
      - _id
      - user_name
      - status
      - last_saved
    additionalProperties: false

required:
  - collection_name
  - collection_id
additionalProperties: false